packages/storage/src/components/UpdateUserPhotoModal/index.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Button, RenderIf, Spinner } from 'react-rainbow-components';
+import { useFirebaseApp, useCurrentUserState } from '@rainbow-modules/firebase-hooks';
+import { nanoid } from 'nanoid';
+import {
+    getStorage,
+    ref,
+    getDownloadURL,
+    uploadBytesResumable,
+    deleteObject,
+} from 'firebase/storage';
+import { updateProfile } from 'firebase/auth';
+import EditRemoveDialog from './editRemoveDialog';
+import {
+    StyledModalEdit,
+    StyledModalFooter,
+    StyledButtonFocus,
+    StyledCropperContainer,
+    StyledCropper,
+    StyledModal,
+    Loading,
+    StyledIcon,
+} from './styled';
+import RotateLeftIcon from './rotateLeftIcon';
+
+export default function UpdateUserPhotoModal(props) {
+    const { path, photo, bucket, isOpen, onRequestClose, avatarInitials, handlePhotoUrl } = props;
+
+    const app = useFirebaseApp();
+    const { user, reload } = useCurrentUserState() || {};
+
+    const pathProp = path || `users/${user ? user.uid : ''}/profilePhoto`;
+    const bucketProp = bucket || `${app.options.storageBucket}`;
+
+    const [photoURL, setPhotoURL] = useState(photo);
+    const [isOpenEdit, setIsOpenEdit] = useState(false);
+    const cropperRef = useRef(null);
+    const [loading, setLoading] = useState(false);
+
+    const handleOnClickEdit = () => {
+        onRequestClose();
+        setIsOpenEdit(true);
+    };
+
+    const onCrop = async () => {
+        const imageElement = cropperRef ? cropperRef.current : null;
+        const cropper = imageElement ? imageElement.cropper : null;
+        const storage = getStorage(app, bucketProp);
+        const photoRef = ref(storage, `${pathProp}/${nanoid()}`);
+        const blob = await fetch(cropper.getCroppedCanvas().toDataURL()).then((r) => r.blob());
+
+        const uploadTask = uploadBytesResumable(photoRef, blob);
+        uploadTask.on(
+            'state_changed',
+            (snapshot) => {
+                const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+                if (progress !== 100) {
+                    setLoading(true);
+                }
+            },
+            (error) => {
+                console.log(error);
```

Comment:
check with @TahimiLeonBravo how handle errors

Reply:
@LeandroTorresSicilia @adrianestevez1 
This is the design for the errors
https://www.figma.com/file/bMnIxYEbFqybEemecEYHLt/rainbow-modules?node-id=150%3A1036&t=yDG1pijxXNgFmCTy-1